CA-288640: Silence xen_watchdog spam

The Xen watchdog emits a warning during shutdown. This warning comes about
because systemd uses the watchdog late at shutdown to avoid hangs preventing a
reboot. It opens the watchdog with O_CLOEXEC and sets a timeout, then execs the
systemd-shutdown binary which reopens the watchdog and completes the final
shutdown tasks. But the xen watchdog emits a warning when when it is closed.

The upstream version of xen_wdt has been rewritten to use the core watchdog
infrastructure and this message was removed as part of that work. Just silence
the warning for now and replace this patch with the following upstream
backports when updating the kernel:
b6c84c9ff3f48a476451eceb5e8478b8aadda8dc
18cffd68e0c44fc09817ea2fb66bebe8a9188093
c41fe7cdbd548121a59323a26d45e20ff08cc2eb

diff --git a/drivers/watchdog/xen_wdt.c b/drivers/watchdog/xen_wdt.c
index cf0e650..e13fa6d 100644
--- a/drivers/watchdog/xen_wdt.c
+++ b/drivers/watchdog/xen_wdt.c
@@ -137,7 +137,6 @@ static int xen_wdt_release(struct inode *inode, struct file *file)
 	if (expect_release)
 		err = xen_wdt_stop();
 	else {
-		pr_crit("unexpected close, not stopping watchdog!\n");
 		xen_wdt_kick();
 	}
 	is_active = err;
